addons/crm_iap_enrich/models/crm_lead.py

```diff
@@ -53,7 +53,7 @@ def iap_enrich(self, from_cron=False):
         batches = [self[index:index + 50] for index in range(0, len(self), 50)]
         for leads in batches:
             lead_emails = {}
-            with self._cr.savepoint():
+            with self._cr.savepoint():  # TODO savepoint in loop
```

Comment:
@lobaudoux See above

Reply:
Agreed with @xmo-odoo, in its current state, this savepoint isn't useful.

I assume the original intent was to let the rest of the batch be processed in case one of them crashes because of a serialization failure, but it isn't properly implemented as it is, it should be:
```diff
-            with self._cr.savepoint():
-                try:
-                    (...)
+            try:
+                with self._cr.savepoint():
+                    (...)
```
If you want to remove the savepoint altogether, fine by me :+1:.